firstvoices/backend/tasks/import_job_tasks.py

```diff
@@ -263,31 +241,45 @@ def import_video_resource(import_job, video_data, dictionary_entry_data, dry_run
         import_job=import_job.id,
     ).import_data(dataset=video_data, dry_run=dry_run)
 
+    video_filename_map = {}
+
     # Adding video ids
     if video_import_result.totals["new"]:
-        video_lookup = {
-            row["video_filename"]: row["id"]
-            for row in video_data.dict
-            if row.get("video_filename")
-        }
-        dictionary_entry_data.append_col(
-            [""] * len(dictionary_entry_data), header="related_videos"
-        )
-        related_video_col_index = dictionary_entry_data.headers.index("related_videos")
-        for i, row in enumerate(dictionary_entry_data.dict):
-            video_filename = row.get("video_filename")
-            related_id = video_lookup.get(video_filename, "")
-            row_list = list(dictionary_entry_data[i])
-            row_list[related_video_col_index] = related_id  # comma separated string
-            dictionary_entry_data[i] = tuple(row_list)
+        for row in video_data.dict:
+            video_filename_map[row["video_filename"]] = row["id"]
 
-    return video_import_result, dictionary_entry_data
+    return video_import_result, video_filename_map
 
 
-def import_dictionary_entry_resource(import_job, dictionary_entry_data, dry_run):
+def import_dictionary_entry_resource(
+    import_job,
+    dictionary_entry_data,
+    audio_filename_map,
+    img_filename_map,
+    video_filename_map,
+    dry_run,
+):
     """
     Imports dictionary entries and returns the import result.
     """
+
+    for filename_key, column_name, media_map in [
+        ("audio_filename", "related_audio", audio_filename_map),
+        ("img_filename", "related_images", img_filename_map),
+        ("video_filename", "related_videos", video_filename_map),
+    ]:
+        if media_map:
+            dictionary_entry_data.append_col(
+                [""] * len(dictionary_entry_data), header=column_name
+            )
+            col_index = dictionary_entry_data.headers.index(column_name)
+            for i, row in enumerate(dictionary_entry_data.dict):
+                filename = row.get(filename_key)
+                related_id = media_map.get(filename, "")
+                row_list = list(dictionary_entry_data[i])
+                row_list[col_index] = related_id
+                dictionary_entry_data[i] = tuple(row_list)
+
```

Comment:
This works, but it loops through the list of dictionary entries 3 times. It would be more efficient to loop through one time, and look up the three filenames as needed.

Reply:
Good idea, I'll try it and see if I can further optimize it.